app/models/proposal.rb

```diff
@@ -9,15 +9,21 @@ class Proposal < ActiveRecord::Base
       event :approve, :transitions_to => :approved
       event :reject, :transitions_to => :rejected
       event :restart, :transitions_to => :pending
+      event :cancel, :transitions_to => :cancelled
     end
     state :approved do
       event :restart, :transitions_to => :pending
+      event :cancel, :transitions_to => :cancelled
     end
     state :rejected do
       # partial approvals and rejections can't break out of this state
       event :partial_approve, :transitions_to => :rejected
       event :reject, :transitions_to => :rejected
       event :restart, :transitions_to => :pending
+      event :cancel, :transitions_to => :cancelled
+    end
+    state :cancelled do
+      event :partial_approve, :transitions_to => :cancelled
```

Comment:
Can a canceled proposal be restarted?

Also, let's get rid of rejected while you're working on this